lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1667,6 +1667,60 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// Temporary: remove unused function args
+// The array arguements are impelemented as local memory ops in handshake
+// dialect, and the original arguements have no use. This pass temporary removes
+// these used args to let the simulator work properly by reading external data
+// files.
+namespace {
+
+struct HandshakeCleanFuncArgPass
+    : public HandshakeCleanFuncArgBase<HandshakeCleanFuncArgPass> {
+  void runOnOperation() override {
+    auto m = getOperation();
+    for (auto funcOp : m.getOps<handshake::FuncOp>()) {
+      OpBuilder rewriter(funcOp);
+
+      // Get function arguments
+      std::vector<unsigned int> argIndices;
+      llvm::SmallVector<mlir::Type, 8> argTypes;
+      auto i = 0;
+      for (auto &arg : funcOp.getArguments()) {
+        auto type = arg.getType();
+        if (arg.use_empty())
+          argIndices.push_back(i);
+        else if (arg.hasOneUse()) {
+          auto sinkOp =
+              dyn_cast<handshake::SinkOp>(arg.use_begin()->getOwner());
+          if (sinkOp) {
+            sinkOp.erase();
+            argIndices.push_back(i);
+          } else
+            argTypes.push_back(type);
+        } else
+          argTypes.push_back(type);
+        i++;
+      }
+
+      // Get function results
+      llvm::SmallVector<mlir::Type, 8> resTypes;
+      for (auto arg : funcOp.getType().getResults())
+        resTypes.push_back(arg);
+
+      auto funcType = rewriter.getFunctionType(argTypes, resTypes);
+      auto originalNumArgs = i;
+      impl::eraseFunctionArguments(funcOp, argIndices, originalNumArgs,
```

Comment:
Can you just use `funcOp.eraseArguments(argIndices)`? I think in this case you won't need to track `i`, `argTypes`, or `resTypes`.